product_sticker/models/product_sticker.py

```diff
@@ -0,0 +1,176 @@
+        return category_domain
+
+    @api.model
+    def _build_sticker_domain_attributes(self, attributes=None, attribute_values=None):
+        """Build domain for attributes and attribute values"""
+        attribute_domain = [
+            ("product_attribute_id", "=", False),
+            ("product_attribute_value_id", "=", False),
+        ]
+        if attribute_values:
+            full_attributes = attributes | attribute_values.mapped("attribute_id")
+            attribute_domain = expression.OR(
+                [
+                    attribute_domain,
+                    expression.OR(
+                        [
+                            [
+                                (
+                                    "product_attribute_value_id",
+                                    "in",
+                                    attribute_values.ids,
+                                )
+                            ],
+                            expression.AND(
+                                [
+                                    [("product_attribute_value_id", "=", False)],
+                                    [
+                                        (
+                                            "product_attribute_id",
+                                            "in",
+                                            full_attributes.ids,
+                                        )
+                                    ],
+                                ]
+                            ),
+                        ]
+                    ),
+                ]
+            )
+        elif attributes:
+            attribute_domain = expression.OR(
+                [
+                    attribute_domain,
+                    expression.AND(
+                        [
+                            [("product_attribute_value_id", "=", False)],
+                            expression.OR(
+                                [
+                                    [("product_attribute_id", "in", attributes.ids)],
+                                    [("product_attribute_id", "=", False)],
+                                ]
+                            ),
+                        ]
+                    ),
+                ]
+            )
+        return attribute_domain
+
+    def _build_sticker_domain(self, **kwargs):
+        company_domain = self._build_sticker_domain_company()
+        category_domain = self._build_sticker_domain_category(kwargs.get("categories"))
+        attribute_domain = self._build_sticker_domain_attributes(
+            kwargs.get("attributes"), kwargs.get("attribute_values")
+        )
+        return [company_domain, category_domain, attribute_domain]
+
+    @api.model
+    def get_stickers(self, **kwargs):
+        """Get stickers for given categories, attributes and attribute values"""
+        sticker_domain = expression.AND(self._build_sticker_domain(**kwargs))
+        return self.sudo().search(sticker_domain)
```

Comment:
Why `sudo()`?

Also, remember to _not_ return recordsets in public methods (explained above).

Reply:
To allow fetch all images from multiple companies

Reply:
And what's the use case? 🤔